File: Data Structures & Algorithms/subarray-product-less-than-k/submission-1.java
class Solution {
    public int numSubarrayProductLessThanK(int[] nums, int k) {
        int count = 0;
        int winStart = 0;
        int prod = 1;
        if(k <=1) return 0;
        for (int winEnd = 0; winEnd < nums.length; winEnd++) {
            prod = nums[winEnd] * prod;
            while (prod >= k) {
                prod = prod / nums[winStart];
                winStart++;
            }
            count += winEnd - winStart + 1;
        }
        return count;
    }
}

File: Data Structures & Algorithms/remove-element/submission-1.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        n = len(nums)
        L = 0

        for R in range(n):
            if(nums[R] != val):
                nums[L] = nums[R]
                L += 1
        # [2,2,2,3]
        #  R
        #  L
        return L